### PR TITLE
Add README

### PR DESCRIPTION
**Pull request by tiwanari**
I'm writing README.md for drive. I want to discuss what we should include (or exclude) in it.

### CODE REVIEW

**Comment by tiwanari** [comment]
It's time to make this PR finished and this repository open, isn't it? :tada:

**Comment by tiwanari** [comment]
Why Travis says 'fails'?

**Comment by mandaiy** [comment]
Do you have any solid reasons you have merged this instead of rebasing?
Rebasing is preferable in a case where the mainline has not changed the same files.

And in my opinion, I'm sorry that, we should not open (make announcement) this repository
unless it has our pile robot features.

**Comment by tiwanari** [comment]
Thanks @myusak ,

> Do you have any solid reasons you have merged this instead of rebasing?

No, actually. I just want to discuss README by using a PR not an issue. I agree with you and will rebase it after getting approvals. Is there any comment on README itself?

> And in my opinion, I'm sorry that, we should not open (make announcement) ...

Oh, I think it is also okay to wait for the feature of our original robots but opening it just with the nxt feature is also okay and valuable not only for others but also for us because anyone can check our codes and may find some bugs. What do you think?

Anyway, regardless of its decision, I think we have to consider whether there is any problem if we open NxtDrive alone. If no, we can do the same refactorings to other product flavors and - if it is okay - open it.

File: README.md
Comment (mandaiy) [inline README.md]: 英文として, 「（世間一般の ） Android アプリケーションは」という主語になる気がします。
また "provide sth to" は不定詞をとれないような。

File: README.md
Comment (mandaiy) [inline README.md]: "visual programming" には冠詞いらない気が

File: README.md
Comment (mandaiy) [inline README.md]: Don't we need our website?

**Review by mandaiy — APPROVED** [review]
Added comments.
But I'm not good at English, so if you feel my comments are garbage, you can go ahead.

**Comment by mandaiy** [comment]
Hmm, ok, I understood your opinion.
But I feel our code still needs refactoring.
So I cannot agree with `It's time to make this PR finished and this repository open`.

**Comment by tiwanari** [comment]
Thanks, 

> But I feel our code still needs refactoring.

This is what I wanted to know 👍 So, what features we should refactor? We should make a milestone.

**Comment by mandaiy** [comment]
At least we should resolve:
- #43 
- #40 
- #24 
- making sure that all files have copyright notice
- making sure that all files are applied our formatter
- (#3 seems to be resolved)

The new github feature, [project](https://github.com/PileProject/drive/projects) could boost us for release, you should try it.

In addition to them I hope to resolve:,
- #45
- #33 
- bump [SquiDB](https://github.com/yahoo/squidb) version up to 3.x 

For further improvements, I really want the feature of compiling blocks before execution.
That is essentially to generate an `ArrayList`, each of whose element is a pair of block and next address,
that array might be like:

```
+------+-----+-------+---------+-----+-----+-------+-------------+------+
|      |  0  |   1   |    2    |  3  |  4  |   5   |      6      |  7   |
+------+-----+-------+---------+-----+-----+-------+-------------+------+
| Kind | Seq | While | If      | Seq | Seq | EndIf | EndWhile    | TERM |
|      | 1   | 2     | True:3  | 5   | 5   | 6     | overzero:2  |      |
| Info |     |       | False:4 |     |     |       | otherwise:7 |      |
|      |     |       |         |     |     |       | counter     |      |
+------+-----+-------+---------+-----+-----+-------+-------------+------+
```

One execution is done by sequentially accessing the array from 0-th index, and the next block address can be fetched by the associated information.
I believe that kind of process is superior to the current solution.
Obviously this needs more consideration, so you can ignore it now (If I can come up with a neat solution, I will make an issue or a PR).

**Comment by mandaiy** [comment]
And do you have a concrete deadline?

**Comment by tiwanari** [comment]
Hi, thanks for listing up stuff 🙇 
I made a project for [`Release`](https://github.com/PileProject/drive/projects/1). 

> And do you have a concrete deadline?

We have a plan to update our website to advertise our products and kick it into high gear from around Dec. 20. So, I think it is good to make this repository open at least by the end of this year.

**Comment by tiwanari** [comment]
Hi @myusak @amiq11 

I came up with adding some screenshots into README and I'll prepare them. 
Do you have any more suggestions?:)

**Comment by tiwanari** [comment]
Hi @makotoshimazu @myusak 

Sorry for sending many PRs at one time but please check this PR, too.
This PR has not many changes. 
I added some screenshots and updated README according to the images.

If you give me an approval, I will rebase after merging the current `develop` to `master`.
Thank you in advance.

**Review by mandaiy — APPROVED** [review]

**Comment by tiwanari** [comment]
Thank you for your check, @myusak 👍 

I will rebase it if @makotoshimazu doesn't refuse this change by next Saturday.

**Comment by tiwanari** [comment]
OK, so let me merge this:)

**Comment by tiwanari** [comment]
Sorry, I made the wrong things on this PR... but [this commit](https://github.com/PileProject/drive/commit/71503bed2056d8b5faf578ebcd964ec402d6ceda) is what I did on this PR. Please ignore the current PR's changes.

**Pull request closed without merging**
